src/components/smart_objects/src/array_schema_item.cc

```diff
@@ -90,6 +92,31 @@ errors::eType CArraySchemaItem::validate(
   return errors::OK;
 }
 
+bool CArraySchemaItem::filterInvalidEnums(
+    SmartObject& Object,
+    const utils::SemanticVersion& MessageVersion,
+    rpc::ValidationReport* report__) {
+  if (SmartType_Array == Object.getType()) {
```

Comment:
I would use reverse if end early exist. Just as suggestion.

Reply:
Are you just suggesting the operands be swapped? I'm a bit confused by this comment

Reply:
Consider following approach : 
```cpp
if (SmartType_Array != Object.getType()) { 
  return false;
  // This block may be removed within removing Object.getType(
}

auto array = Object.asArray();
auto elements_to_remove = std::find_if(array->begin(), array->end(), []() {
  mElementSchemaItem->filterInvalidEnums(...)
});
for (element: elements_to_remove) {
  array.eraze(element);
}
// min size staff using elements_to_remove.size()
}
```
I didn't compile it, so there could be issues with syntax. 
 
I see this benefits : 
 - No nested structures 
 - No iterator crutches.

I agree that it probably less efficient then previous one. But we do not expect huge amount of elements in `Object`, do we?